dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/TrackedEntityOperationParamsMapper.java

```diff
@@ -106,9 +130,28 @@ public TrackedEntityQueryParams map(TrackedEntityOperationParams operationParams
         .setIncludeDeleted(operationParams.isIncludeDeleted())
         .setPotentialDuplicate(operationParams.getPotentialDuplicate());
 
+    validateGlobalSearchParameters(params);
+
     return params;
   }
 
+  private List<TrackedEntityType> filterAndValidateTrackedEntityTypes(
+      User user, TrackedEntityType trackedEntityType, Program program) throws BadRequestException {
+    List<TrackedEntityType> trackedEntityTypes = new ArrayList<>();
+    if (trackedEntityType == null) {
```

Comment:
If program is not null, should we still define the `trackedEntityTypes`?

Reply:
when the program is not null we still need to do it, but that's a good point, because we don't need to do it if a `trackedEntityType` is present in the request.

Reply:
If Program is not null, shouldn't be somoething like this?
`trackedEntityTypes = List.of(program.getTrackedEntityType)`

Reply:
I did it this way because it's how it's done before this refactor.
When no `trackedEntityType` is specified, it gets all the ones accessible by the user. Is there a reason why we should use the one of the program instead?

 https://github.com/dhis2/dhis2-core/blob/master/dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/DefaultTrackedEntityService.java#L472